### ISSUE TITLE
Pagination does not make request to server for more pages 

### ISSUE THREAD

**Issue by Rytiggy** [eclipse-pass/pass-ui#368]
Currently the tables only pull the first 10 submissions / grants and pagination does not pull the next 10. 

Implimenting models-table-server-paginated should fix this
http://onechiporenko.github.io/ember-models-table/v.2/docs/classes/Components.ModelsTableServerPaginated.html

![screen shot 2018-05-25 at 2 05 26 pm](https://user-images.githubusercontent.com/17414914/40559423-b92258d0-6024-11e8-8abf-b90a51271ca2.png)
![screen shot 2018-05-25 at 2 05 22 pm](https://user-images.githubusercontent.com/17414914/40559424-b94b3ad4-6024-11e8-8723-dfd84286b7f7.png)
![screen shot 2018-05-25 at 2 05 06 pm](https://user-images.githubusercontent.com/17414914/40559425-b971e95e-6024-11e8-96f8-2b6936b374d0.png)

**Comment by jabrah** [comment]
There is currently a naive search implementation in the UI which does not do paging. However, I didn't realize that the default behavior of `Store.query` was to return only 10 objects. This probably explains this ticket. At a later date we will have to get paging to work.

**Comment by jabrah** [comment]
Note for later impl: 
see `jabrah/dev/server-paged-tables`
https://ember-twiddle.com/3177ccf0b6c8a4656c9724cd8aa4f83a?fileTreeShown=false&numColumns=2&openFiles=templates.index.hbs%2Ccomponents.my-table.js

**Comment by jabrah** [comment]
Starting to add manual paging to the various data tables: https://github.com/jabrah/pass-ember/tree/refactor/server-side-paging

This branch removes controls from the `ember-models-table`'s in appropriate routes and uses custom controls tied to controller actions to be able to hit Elasticsearch when needed

**Comment by jabrah** [comment]
Note: Server side paging is currently stalled because of _filtering_ and _sorting_

There is information displayed in all tables that require paging that does not come from the primary table object. For example, the Submissions table displays a row for each submission, but data comes from the submissions Publication object, associated grant objects, etc. Paging is implemented in the UI by issuing new search requests for each page. Filtering and sorting require queries on properties that are not present on the desired return object.

In the submissions table, when you filter for a word in publication title, you really want to return the submission objects that have a publication object that has the matching word. In the current ES, if you blindly do a query for the `title` property with the desired word, it will return the matching publication object, but there is currently no way on the server side to then return the submission it belongs to. This same issue is relevant to filtering and sorting.

_Possible work around?_
Would it be possible/feasible to add properties from related objects onto related objects in the index? For example when indexing a Submission, pull in its related Publication and add a `publication_filter` property that can be searched for the purpose of filtering and sorting. The property name would have to be distinct so that it can be explicitly ignored in the returned submission object (see how autocomplete `_suggest` fields are ignored).

**Comment by markpatton** [comment]
Yes. You would have to copy properties from child objects to parent objects. This requires custom PASS indexer support. We would have to figure out a reasonable scheme. But this would require a lot of idiosyncratic work on the index. First we should be careful to only keep filtering and sort functionality that is required.

**Comment by markpatton** [comment]
Pagination requests are now made when populating tables.
